src/attribute.h

```diff
@@ -0,0 +1,26 @@
+#ifndef ATTRIBUTE_H
+#define ATTRIBUTE_H
+
+#include <string>
+#include <cstdint> //uint8_t, uint32_t
+
+class Attribute
+{
+    public:
+        Attribute(uint8_t type);
+        virtual ~Attribute();
```

Comment:
In C++11, if you are satisfied with default implementation of some methods (constructors, assignments, destructors) you can mark them `= default` to state explicitly that you want it generated by the compiler. In your case:
```c++
virtual ~Attribute() = default;
```
And that's all, you don't need the definition.